modules/homebrew.nix

```diff
@@ -1,219 +1,751 @@
+  # Option values and descriptions for Brewfile entries are sourced/derived from:
+  #   * `brew` manpage: docs.brew.sh/Manpage
+  #   * `brew bundle` source files:
+  #     * github.com/Homebrew/homebrew-bundle/blob/master/lib/bundle/dsl.rb
+  #     * github.com/Homebrew/homebrew-bundle/blob/master/lib/bundle/{cask,brew,tap}_installer.rb
+  #     * github.com/Homebrew/homebrew-bundle/blob/master/spec/bundle/{cask,brew,tap}_installer_spec.rb
+
+  onActivationOptions = { config, ... }: {
+    options = {
+      cleanup = mkOption {
+        type = types.enum [ "none" "uninstall" "zap" ];
+        default = "none";
+        example = "uninstall";
+        description = ''
+          This option manages what happens to formulae installed by Homebrew, that aren't present in
+          the Brewfile generated by this module, during <command>nix-darwin</command> system
+          activation.
+
+          When set to <literal>"none"</literal> (the default), formulae not present in the generated
+          Brewfile are left installed.
+
+          When set to <literal>"uninstall"</literal>, <command>nix-darwin</command> invokes
+          <command>brew bundle [install]</command> with the <command>--cleanup</command> flag. This
+          uninstalls all formulae not listed in generate Brewfile, i.e.,
+          <command>brew uninstall</command> is run for those formulae.
+
+          When set to <literal>"zap"</literal>, <command>nix-darwin</command> invokes
+          <command>brew bundle [install]</command> with the <command>--cleanup --zap</command>
+          flags. This uninstalls all formulae not listed in the generated Brewfile, and if the
+          formula is a cask, removes all files associated with that cask. In other words,
+          <command>brew uninstall --zap</command> is run for all those formulae.
+
+          If you plan on exclusively using <command>nix-darwin</command> to manage formulae
+          installed by Homebrew, you probably want to set this option to
+          <literal>"uninstall"</literal> or <literal>"zap"</literal>.
+        '';
+      };
+      update = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Homebrew to auto-update itself and all formulae during
+          <command>nix-darwin</command> system activation. The default is <literal>false</literal>
+          so that repeated invocations of <command>darwin-rebuild switch</command> are idempotent.
+
+          If you prefer this behavior you many also want to consider enabling
+          ${mkDocOptionLink "homebrew.global.noAutoUpdate"}.
+        '';
+      };
+      upgrade = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Homebrew to upgrade outdated formulae and Mac App Store apps during
+          <command>nix-darwin</command> system activation. The default is <literal>false</literal>
```

Comment:
I'm a bit unsure about  brews ability to upgrade `Mac App Store apps`. Are you implicitly referring to mas here?

Reply:
Well `brew bundle` can initiate upgrading of apps installed from the Mac App Store that are listed in the Brewfile with `mas` entries. When the `--no-upgrade` flag is passed to `brew bundle` it doesn't ask `mas` to upgrade the apps listed in the Brewfile that are outdated.

Reply:
Oh, nice. TIL.